python/tskit/combinatorics.py

```diff
@@ -28,12 +28,133 @@
 import functools
 import heapq
 import itertools
+import json
+import random
 
+import attr
 import numpy as np
 
 import tskit
 
 
+@attr.s(eq=False)
+class TreeNode:
+    """
+    Simple linked tree class used to generate tree topologies.
+    """
+
+    parent = attr.ib(default=None)
+    children = attr.ib(factory=list)
+    label = attr.ib(default=None)
+
+    @staticmethod
+    def random_binary_tree(leaf_labels, rng):
```

Comment:
Should we document the arguments, leaf_labels and rng?

Reply:
Done. I've not done the formal sphinx params as this isn't part of the public API.